### PR TITLE
Part 2 v6 release: update annotated cnv files

### PR DESCRIPTION
**Pull request by kgaonkar6**
#### What scientific question is your analysis addressing?
Updating  `focal-cn-prepration-files` to run for OT. 

#### What was your approach?
- updated `pbta-histologies.tsv` to `histologies.tsv` 
- used `guess=100000` because without it tumor_ploidy was `NA`
- updated cnvkit and  controlfreec filename
- retained `neutral` calls
- added `deep deletion`

#### What GitHub issue does your pull request address?
PediatricOpenTargets/ticket-tracker#60

#### Which areas should receive a particularly close look?
- The CNVs are annotated by genes, and any cnv not annotated by cnv are removed [here]( https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/f7b092837d4b848c167e979f04c677cafb3b65a1/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L30) does that sound ok? 

- `analyses/focal-cn-file-preparation/run-prepare-cn-OpenTarget.sh` was added to only run scripts to annotate cnvs to genes and not perform chromosome arm level annotation which takes up a long time and resources.

#### Is there anything that you want to discuss further?
The XY chromosomes don't have amplification coded in, should we look into this ?

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes

#### What types of results are included (e.g., table, figure)?
tables

#### What is your summary of the results?
```
> histology %>% filter(Kids_First_Biospecimen_ID %in% focal$biospecimen_id) %>% pull(cohort) %>% table()
.
GMKF PBTA 
 200  652 
```
200 out of 336 GMKF and 652 out of 940 PBTA WGS samples are retained in the consensus cnv calls.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/focal-cn-file-preparation/.gitignore [hunk 1]
```diff
@@ -3,3 +3,12 @@ annotation_files/txdb_from_gencode.v27.gtf.db
 
 # Some GISTIC files are too large to be committed
 gistic-results
+
+# ignore large files
+results/cnvkit_annotated_cn_autosomes.tsv.gz
+results/controlfreec_annotated_cn_autosomes.tsv.gz
+
+# corrsponding xy for the above large files
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/.gitignore]:
```suggestion
# corresponding xy for the above large files
```

File: analyses/focal-cn-file-preparation/04-prepare-cn-file.R [hunk 2]
```diff
@@ -260,11 +261,17 @@ if (xy_flag) {
...
 
   # Merge and annotated no X&Y
   sex_chrom_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_sex_chrom,
-                                                      txdb_exons = tx_exons)
+                                                      txdb_exons = tx_exons) %>%
+    # mark possible deep loss in sex chromosome
+    dplyr::mutate(status = dplyr::case_when(
+      copy_number == 0  ~ "deep deletion",
+      TRUE ~ as.character(status)
+    ))
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/04-prepare-cn-file.R:273]: I think we will come back to this, as I think due to the nature of the sex chr not always being diploid (XY), the losses are capturing broad and not just focal losses, so adding a note here.

**Review by jharenza — APPROVED** [review]
just one typo change and good to go for now

**Comment by kgaonkar6** [comment]
Thanks for the review @jharenza! I'll make a discussion ticket about the cnv statuses in sex chr.

**Comment by kgaonkar6** [comment]
The ticket about focal and gene annotation is in: https://github.com/PediatricOpenTargets/ticket-tracker/issues/86. Should I merge this now?

**Pull request merged**